onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/RetainUpFromPolygon.java

```diff
@@ -0,0 +1,94 @@
+package org.onebusaway.gtfs_transformer.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import java.io.Serializable;
+
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.geom.*;
+
+import org.onebusaway.csv_entities.schema.annotations.CsvField;
+import org.onebusaway.gtfs.model.IdentityBean;
+import org.onebusaway.gtfs.model.Stop;
+import org.onebusaway.gtfs.serialization.GtfsEntitySchemaFactory;
+import org.onebusaway.gtfs.services.GtfsMutableRelationalDao;
+import org.onebusaway.gtfs_transformer.factory.EntityRetentionGraph;
+import org.onebusaway.gtfs_transformer.services.GtfsTransformStrategy;
+import org.onebusaway.gtfs_transformer.services.TransformContext;
+import org.slf4j.Logger;
+
+import org.slf4j.LoggerFactory;
+
+public class RetainUpFromPolygon implements GtfsTransformStrategy {
+    private final Logger log = LoggerFactory.getLogger(RetainUpFromPolygon.class);
+
+    @CsvField(optional = false)
+    private String polygon;
+
+    public void setPolygon(String polygon) {
+        this.polygon = polygon;
+    }
+
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public void run(TransformContext transformContext, GtfsMutableRelationalDao gtfsMutableRelationalDao) {
+        Geometry geometry = buildPolygon(polygon);
```

Comment:
Move this into the setter and check if it's valid and non-empty there. Throw an exception immediately if it isn't.